src/Compilers/CSharp/Portable/Syntax/SyntaxFormatter.cs

```diff
@@ -127,8 +129,27 @@ public override SyntaxToken VisitToken(SyntaxToken token)
             }
         }
 
+        private SyntaxToken GetNextRelevantToken(SyntaxToken token)
+        {
+            // get next token, skipping zero width tokens except for end-of-directive tokens
+            var nextToken = token.GetNextToken(
+                t => SyntaxToken.NonZeroWidth(t) || t.Kind() == SyntaxKind.EndOfDirectiveToken, 
+                t => t.Kind() == SyntaxKind.SkippedTokensTrivia);
+
+            if (_consideredSpan.Contains(nextToken.FullSpan))
```

Comment:
I added this to stop the syntax normalizer from considering tokens outside of the sub tree, so you get the same behavior whether your root is parented or not.  It was causing some edge-case weirdness when you were taking a sub-tree out of an existing tree, normalizing it and using it elsewhere. If intersect is used, then it will consider tokens on the edge, outside of the sub-tree when deciding on new-lines and whitespace, and yet those tokens don't appear in the output.